### PR TITLE
Add temporary range operator @..

### PR DESCRIPTION
**Pull request by jroimartin**
Implement #4958

### CODE REVIEW

File: libr/core/cmd.c [hunk 1]
```diff
@@ -1741,6 +1741,44 @@ static int r_core_cmd_subst_i(RCore *core, char *cmd, char *colon) {
...
+				}
+				for (i = 0; tovars[i] != NULL; i++) {
+					curto[i] = r_config_get_i (core->config, tovars[i]);
+				}
+
+				// set new ranges
+				for (i = 0; fromvars[i] != NULL; i++) {
+					r_config_set_i (core->config, fromvars[i], from);
+				}
+				for (i = 0; tovars[i] != NULL; i++) {
```
Comment (radare) [inline libr/core/cmd.c]: unnecessary != NULL

File: libr/core/cmd.c [hunk 2]
```diff
@@ -1754,6 +1792,16 @@ if (addr != UT64_MAX) {
 				} else ret = 0;
```
Comment (radare) [inline libr/core/cmd.c]: missing braces {}

File: libr/core/cmd.c [hunk 3]
```diff
@@ -1741,6 +1741,44 @@ static int r_core_cmd_subst_i(RCore *core, char *cmd, char *colon) {
 			}
 			//ret = -1; /* do not run out-of-foreach cmd */
 		} else {
+			bool tmpseek = false;
+			char *fromvars[] = { "anal.from", "diff.from", "graph.from", "io.buffer.from",
```
Comment (radare) [inline libr/core/cmd.c]: this is const char *

File: libr/core/cmd.c [hunk 4]
```diff
@@ -1741,6 +1741,44 @@ static int r_core_cmd_subst_i(RCore *core, char *cmd, char *colon) {
...
+				 "lines.from", "search.from", "zoom.from", NULL };
+			char *tovars[] = { "anal.to", "diff.to", "graph.to", "io.buffer.to",
```
Comment (radare) [inline libr/core/cmd.c]: same

File: libr/core/cmd.c [hunk 5]
```diff
@@ -1741,6 +1741,44 @@ static int r_core_cmd_subst_i(RCore *core, char *cmd, char *colon) {
...
+				"lines.to", "search.to", "zoom.to", NULL };
+			ut64 curfrom[R_ARRAY_SIZE(fromvars) - 1], curto[R_ARRAY_SIZE(tovars) - 1];
```
Comment (radare) [inline libr/core/cmd.c]: missing spaces before (

File: libr/core/cmd.c [hunk 6]
```diff
@@ -1741,6 +1741,44 @@ static int r_core_cmd_subst_i(RCore *core, char *cmd, char *colon) {
...
+				if (p == NULL) {
+					eprintf ("Usage: / ABCD @..0x1000 0x3000\n");
+					return false;
+				}
+				*p = '\x00';
+				ut64 from = r_num_math (core->num, range);
+				ut64 to = r_num_math (core->num, p + 1);
+
+				// save current ranges
+				for (i = 0; fromvars[i] != NULL; i++) {
```
Comment (radare) [inline libr/core/cmd.c]: unnecessary != NULL

File: libr/core/cmd.c [hunk 7]
```diff
@@ -1754,6 +1792,16 @@ if (addr != UT64_MAX) {
 				} else ret = 0;
 }
 			}
+
+			if (tmpseek) {
+				// restore ranges
+				for (i = 0; fromvars[i] != NULL; i++) {
```
Comment (radare) [inline libr/core/cmd.c]: unnecessary != NULL

**Comment by jroimartin** [comment]
Applied suggested changes

**Pull request merged**